applications/nrf5340_audio/src/audio_module/src/amod_api.h

```diff
@@ -0,0 +1,492 @@
+	/* Module state undefined */
+	AMOD_STATE_UNDEFINED = 0,
+
+	/* Module is in the configured state */
+	AMOD_STATE_CONFIGURED,
+
+	/* Module is in the running state */
+	AMOD_STATE_RUNNING,
+
+	/* Module is in the stopped state */
+	AMOD_STATE_STOPPED
+};
+
+/**
+ * @brief Private pointer to a modules functions.
+ */
+struct amod_functions {
+	/**
+	 * @brief  Function for opening a module.
+	 *
+	 *
+	 * @param handle  A handle to this module instance
+	 *
+	 * @return 0 if successful, error value
+	 */
+	int (*open)(struct handle *handle, struct amod_configuration *configuration);
+
+	/**
+	 * @brief  Function to close an open module.
+	 *
+	 * @param handle  A handle to this module instance
+	 *
+	 * @return 0 if successful, error value
+	 */
+	int (*close)(struct handle *handle);
+
+	/**
+	 * @brief  Function to set the configuration of a module.
+	 *
+	 * @param handle		 A handle to this module instance
+	 * @param configuration  A pointer to the modules configuration to set
+	 *
+	 * @return 0 if successful, error value
+	 */
+	int (*configuration_set)(struct handle *handle, struct amod_configuration *configuration);
+
+	/**
+	 * @brief  Function to get the configuration of a module.
+	 *
+	 * @param handle		 A handle to this module instance
+	 * @param configuration  A pointer to the modules current configuration
+	 *
+	 * @return 0 if successful, error value
+	 */
+	int (*configuration_get)(struct handle *handle, struct amod_configuration *configuration);
+
+	/**
+	 * @brief Start a module processing data.
+	 *
+	 * @param handle  The handle for the module to start
+	 *
+	 * @return 0 if successful, error value
+	 */
+	int (*start)(struct handle *handle);
+
+	/**
+	 * @brief Stop a module processing data.
+	 *
+	 * @param handle  The handle for the module to stop
+	 *
+	 * @return 0 if successful, error value
+	 */
+	int (*stop)(struct handle *handle);
+
+	/**
+	 * @brief The core data processing function for the module. Can be either an
+	 *		input or output or processor module type.
+	 *
+	 * @param handle	 A handle to this module instance
+	 * @param handle_tx  Pointer to the input audio block or NULL for an input module
+	 * @param block_rx   Pointer to the output audio block or NULL for an output module
+	 *
+	 * @return 0 if successful, error value
+	 */
+	int (*data_process)(struct handle *handle, struct aobj_block *handle_tx,
+			    struct aobj_block *block_rx);
+};
+
+/**
+ * @brief A modules minimum description.
+ */
+struct amod_description {
+	/* The module base name */
+	char *name;
+
+	/* The module type */
+	enum amod_type type;
+
+	/* A pointer to the functions to implment the module */
+	const struct amod_functions *functions;
+};
+
+/**
+ * @brief Module's thread configuration structure.
+ */
+struct amod_thread_configuration {
+	/* Thread stack */
+	k_thread_stack_t *stack;
+
+	/* Thread stack size */
+	size_t stack_size;
+
+	/* Thread priority */
+	int priority;
+
+	/* Pointer to a modules data receiver fifo */
```

Comment:
```suggestion
	/* Pointer to a modules data receiver FIFO*/
```

Same for the ones below